src/lamplib/__main__.py

```diff
@@ -7,22 +7,27 @@
 import parser
 import tasks
 import tasks.run_tests
+from tasks.download import ToolchainDownloader, CuratorDownloader
 from context import Context
 from parser import add_args_to_context
-from tasks.toolchain import get_toolchain_url, fetch_and_install_toolchain
 
+def check_venv(args):
+    if not 'VIRTUAL_ENV' in os.environ and not args.run_global:
+        logging.error('Tried to execute without active virtualenv. If you want to run lamp '
+                      'without a virtualenv, use the --run-global option.')
+        sys.exit(1)
 
 def run_self_test():
     res = subprocess.run(['python3', '-m', 'unittest'],
                          cwd=os.path.dirname(os.path.abspath(__file__)))
     res.check_returncode()
     sys.exit(0)
 
-
 def validate_environment():
     # Check Python version
     if not sys.version_info >= (3, 7):
-        logging.error('Please run this script with Python 3.7 or newer')
+        logging.error('Detected Python version less than 3.7. Please delete '
+                      'the virtualenv and run lamp again.')
```

Comment:
if feasible/easy, let's include `sys.version_info` in the message e.g. "Detected Python version (3.6) less than 3.7....`